src/NosCore.GameObject/Providers/NRunProvider/Handlers/TeleporterHandler.cs

```diff
@@ -0,0 +1,80 @@
+// 
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using NosCore.GameObject.ComponentEntities.Interfaces;
+using NosCore.GameObject.Networking.ClientSession;
+using ChickenAPI.Packets.Enumerations;
+using ChickenAPI.Packets.ClientPackets.Npcs;
+using NosCore.Core;
+using NosCore.GameObject.ComponentEntities.Extensions;
+using NosCore.Core.I18N;
+using NosCore.Data.Enumerations.I18N;
+
+namespace NosCore.GameObject.Providers.NRunProvider.Handlers
+{
+    public class TeleporterHandler : IEventHandler<Tuple<IAliveEntity, NrunPacket>, Tuple<IAliveEntity, NrunPacket>>
+    {
+        public bool Condition(Tuple<IAliveEntity, NrunPacket> item) => item.Item2.Runner == (NrunRunnerType)16;
+
+        public void Execute(RequestData<Tuple<IAliveEntity, NrunPacket>> requestData)
+        {
+            var tp = requestData.ClientSession.Character.MapInstance.Npcs.Find(
+                s => s.VisualId == requestData.Data.Item2.VisualId
+                && s.VisualType == requestData.Data.Item2.VisualType
+                && s.Dialog >= 439 && s.Dialog <= 441);
+
+            if (tp != null)
+            {
+                bool CheckOut(long GoldToPay)
+                {
+                    if (requestData.ClientSession.Character.Gold >= GoldToPay)
+                    {
+                        requestData.ClientSession.Character.RemoveGold(GoldToPay);
+                        return true;
+                    }
+                    requestData.ClientSession.SendPacket(requestData.ClientSession.Character.GenerateSay(
+                        Language.Instance.GetMessageFromKey(LanguageKey.NOT_ENOUGH_MONEY, requestData.ClientSession.Account.Language), SayColorType.Yellow
+                            ));
+                    return false;
+                }
+
+                if (requestData.Data.Item2.Type == 0)
+                {
+                    requestData.ClientSession.ChangeMap(
+                        1, (short)RandomFactory.Instance.RandomNumber(78, 81), (short)RandomFactory.Instance.RandomNumber(114, 118));
+                } else if (requestData.Data.Item2.Type == 1)
+                {
+                    if (CheckOut(1_000))
+                    {
+                        requestData.ClientSession.ChangeMap(
+                            20, (short)RandomFactory.Instance.RandomNumber(7, 11), (short)RandomFactory.Instance.RandomNumber(90, 94));
+                    }
+                } else
+                {
+                    if (CheckOut(2_000))
+                    {
+                        requestData.ClientSession.ChangeMap(
```

Comment:
This can go inside checkout and you can then drop the bool. Just need to add more parameters